doc/build.rst

```diff
@@ -161,6 +161,21 @@ To build with MinGW, type:
 
   cp make/mingw64.mk config.mk; make -j4
 
+Opening a can of worms - building XGBoost library for python for Windows with MinGW-w64
+---------------------------------------------------------------------------------------
+
+Windows versions of python are built with Microsoft proprietary Visual Studio. This has multiple issues:
+
+1. VS is proprietary and commercial software (it may be illegal to install the free version in your org). It is bundled with telemetry. It also has been spotted on inserting telemetry into apps.
```

Comment:
Let's re-phrase this bullet point as follows:

1. VS is proprietary and commercial software. While Microsoft provides a free edition (Community Edition), installing it may be against the policy of your organization. In addition, Visual Studio contains telemetry, as documented in Microsoft Visual Studio Licensing Terms (https://visualstudio.microsoft.com/license-terms/mt736442/). This has raised privacy and security concerns among users and system administrators.

I took out the part about telemetry insertion into apps, since Microsoft has provided a response to the allegation: https://www.reddit.com/r/cpp/comments/4ibauu/visual_studio_adding_telemetry_function_calls_to/.